test-runner/test/exit-code.spec.ts

```diff
@@ -13,32 +13,52 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-import "../lib"
-import { spawnSync } from "child_process";
-import path from 'path';
+
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+import rimraf from 'rimraf';
+import { promisify } from 'util';
+import '../lib';
+
+const removeFolderAsync = promisify(rimraf);
 
 it('should fail', async() => {
-  const result = runTest('one-failure.js');
+  const result = await runTest('one-failure.js');
   expect(result.exitCode).toBe(1);
   expect(result.passed).toBe(0);
   expect(result.failed).toBe(1);
 });
 
 it('should succeed', async() => {
-  const result = runTest('one-success.js');
+  const result = await runTest('one-success.js');
   expect(result.exitCode).toBe(0);
   expect(result.passed).toBe(1);
   expect(result.failed).toBe(0);
 });
 
-function runTest(filePath: string) {
-  const {output, status} = spawnSync('node', [path.join(__dirname, '..', 'cli.js'), path.join(__dirname, 'assets', filePath)]);
-  const passed = (/  (\d+) passed/.exec(output.toString()) || [])[1];
-  const failed = (/  (\d+) failed/.exec(output.toString()) || [])[1];
+it('should access error in fixture', async() => {
+  const result = await runTest('test-error-visible-in-fixture.js');
+  expect(result.exitCode).toBe(1);
+  const data = JSON.parse(fs.readFileSync(path.join(__dirname, 'test-results', 'test-error-visible-in-fixture.txt')).toString());
+  expect(data.message).toContain('Object.is equality');
+});
+
+async function runTest(filePath: string) {
+  const outputDir = path.join(__dirname, 'test-results')
```

Comment:
this will break if tests are run in parallel, which they are in my patch in review. https://github.com/microsoft/playwright/pull/3595

Reply:
Would it make sense to convert runTest into a fixture that depends on the parallelIndex then to isolate output folders?